### DISCUSSION TITLE
Getting the splash screen again and again

### DISCUSSION THREAD

**Discussion by GutHib** [organicmaps/organicmaps#4797] [Q&A]
Quick question: When I open Organic Maps and reaccess it after a while (timespan varies, but let's say, half an hour), I get the green splash screen again, even though the app is still "open" (which is actually quite annoying, because my phone is quite slow and loading takes a while). I there a known reason for this behavior?

Thanks!

**Comment by Jean-BaptisteC** [comment]
Android or IOS ?
Which version of OS and OM version ?

**Comment by biodranik** [comment]
It looks like your Android system kills activities when they are in the background. It is not related to Organic Maps.

You can check your system's settings. You may check some articles about it, for example, this one: https://www.geekdashboard.com/stop-android-killing-apps-background/

**Comment by GutHib** [comment]
Thanks for the replies - I was wondering if it's OS-related, on the other hand, I've never seen anything like it with any of my other apps, ever.

Set battery usage to un-optimized now to see if it makes a difference.

OS is Android Lineage OS. Organic Maps version 2022.12.18-12-FDroid.

**Comment by GutHib** [comment]
Just to be clear - are you suggesting that Android is killing some "activity" (i.e. a spawned sub-process) or Organic Maps itself? Because like I wrote above, the app itself is still "open" (and by that, I mean, it's still up and running when I cycle through background apps). Yet, when I select it, I (may) get the green splash screen again.

Fiddling with battery usage did nothing.

I'm happy to run further tests but would need to know what to look into. What if I download Maps.me to see if it has the same issue - would that be proof enough (i.e. same problem would point to Android as the culprit, whereas no problem would point to an issue with Organic Maps)?

**Reply by Misalf-git** [reply]
> Just to be clear - are you suggesting that Android is killing some \"activity\" (i.e. a spawned sub-process) or Organic Maps itself? Because like I wrote above, the app itself is still \"open\" (and by that, I mean, it's still up and running when I cycle through background apps). Yet, when I select it, I (may) get the green splash screen again.

The "opened apps" visible in the apps switcher are merely sreenshots of what "was running".

The OS decides if processes, which may not do much/anything but sitting on resources, can be killed/closed to free up resources in order to allow other processes to use these, otherwise "wasted" resources, or to simply save energy.

Processes that refuse te be stopped are either rouge, privileged or part of the OS.

**Reply by biodranik** [reply]
Android can destroy activities, but leave the process in the memory. In this case, you may see the splash, but for a very short time only (hot-start). Or, Android can kill the whole app's process. It leads to a cold start.
If an activity was not killed, the app opens immediately. It should be somehow related to Lineage OS.

Also, why didn't you update to the latest Organic Maps version?

**Reply by biodranik** [reply]
There are some related answers here: https://www.reddit.com/r/LineageOS/comments/mykwmh/how_to_stop_lineage_171_killing_background_apps/

**Comment by GutHib** [comment]
To sum it up:
- Splash screen doesn't just show up briefly, so by the above definition, it's a cold start.
- Turning off battery optimization didn't help. Couldn't find any other related settings.
- Updating to the newest OM version didn't help.
- Turning OM into a system app didn't help.
- Fresh boot / closing all other background apps (at least those in the app switcher) didn't help.

Let's say the problem didn't occur in Maps.me - would that be valuable info, or is the final verdict already out (as in: there's nothing that can be done)?

**Reply by biodranik** [reply]
AFAIR, in the system's Developer Settings were some related options. Did you also check them?

Cold start of OM can happen only in two cases:
1. System killed OM for some reason (low memory, system power saving settings, developer settings).
2. OM crashed when it was closed. Could it be your case? Is the map view always stored correctly between cold restarts?

Am I correctly understanding that maps.me has a longer splash only on a cold start?

What is your device model and Lineage OS version?

**Comment by GutHib** [comment]
For a while, I thought that turning off battery manager globally would do the trick, but in the end, that doesn't help either. The question remains if all apps are affected equally and OM is simply the only one where I notice, or if OM is, for some reason, more likely to be shot down than others (e.g. due to the priorities of the operating system and/or something OM-specific).

Map view is stored correctly between cold restarts. Also, I can't remember OM ever crashing when it's on, so I would be very surprised if it always crashes when I'm not looking.

EDIT: Tested Maps.me and it shows the same behavior.

Samsung S5 and lineage 18.1 (based on Android 11).
